pkg/types/cciptypes/commitstore.go

```diff
@@ -0,0 +1,81 @@
+package cciptypes
+
+import (
+	"context"
+	"math/big"
+	"time"
+)
+
+type CommitStoreReader interface {
+	GetExpectedNextSequenceNumber(context context.Context) (uint64, error)
+
+	GetLatestPriceEpochAndRound(context context.Context) (uint64, error)
+
+	// GetCommitReportMatchingSeqNum returns accepted commit report that satisfies Interval.Min <= seqNum <= Interval.Max. Returned slice should be empty or have exactly one element
+	GetCommitReportMatchingSeqNum(ctx context.Context, seqNum uint64, confirmations int) ([]CommitStoreReportWithBlockMeta, error)
+
+	// GetAcceptedCommitReportsGteTimestamp returns all the commit reports with timestamp greater than or equal to the provided.
+	// Returned Commit Reports have to be sorted by Interval.Min/Interval.Max in ascending order.
+	GetAcceptedCommitReportsGteTimestamp(ctx context.Context, ts time.Time, confirmations int) ([]CommitStoreReportWithBlockMeta, error)
+
+	IsDown(ctx context.Context) (bool, error)
+
+	IsBlessed(ctx context.Context, root [32]byte) (bool, error)
+
+	// ChangeConfig notifies the reader that the config has changed onchain
+	ChangeConfig(onchainConfig []byte, offchainConfig []byte) (Address, error)
+
+	OffchainConfig() CommitOffchainConfig
+
+	GasPriceEstimator() GasPriceEstimatorCommit
+
+	EncodeCommitReport(report CommitStoreReport) ([]byte, error)
+
+	DecodeCommitReport(report []byte) (CommitStoreReport, error)
+
+	VerifyExecutionReport(ctx context.Context, report ExecReport) (bool, error)
+
+	GetCommitStoreStaticConfig(ctx context.Context) (CommitStoreStaticConfig, error)
+}
+
+type CommitStoreReportWithBlockMeta struct {
+	BlockMeta
+	CommitStoreReport
+}
+
+type CommitStoreReport struct {
+	TokenPrices []TokenPrice
+	GasPrices   []GasPrice
+	Interval    CommitStoreInterval
+	MerkleRoot  [32]byte
+}
+
+type TokenPrice struct {
+	Token Address
+	Value *big.Int
+}
+
+type GasPrice struct {
+	DestChainSelector uint64
+	Value             *big.Int
+}
+
+type CommitOffchainConfig struct {
+	GasPriceDeviationPPB   uint32
+	GasPriceHeartBeat      time.Duration
+	TokenPriceDeviationPPB uint32
+	TokenPriceHeartBeat    time.Duration
+	InflightCacheExpiry    time.Duration
+}
+
+type CommitStoreStaticConfig struct {
+	ChainSelector       uint64
```

Comment:
is this a chain id? other, non-evm, chains use string identifiers. it's probably best to change this now and adapt from string to int as needed in the evm relayer implementation. @jmank88 do you think?

if we do go this direction, then there are other chain selector fields in other types that should be changed, too.

for example, here is the configuration struct we use to represent a generic chain  https://github.com/smartcontractkit/chainlink-common/blob/5d1fecc0a6992a313c083ed17f53f574588b3af7/pkg/types/relayer.go#L36

Reply:
Nah it is intentionally not a semantic chain id. I think this is fine as-is.
https://github.com/smartcontractkit/chain-selectors